modules/farm/spec/02_messages.md

```diff
@@ -0,0 +1,136 @@
+<!--
+order: 2
+-->
+
+# Messages
+
+## MsgCreatePool
+
+The farm pool can be created by any user via a `MsgCreatePool` message.
+
+```go
+type MsgCreatePool struct {
+    Name           string
+    Description    string
+    LpTokenDenom   string
+    StartHeight    uint64
+    RewardPerBlock sdk.Coins
+    TotalReward    sdk.Coins
+    Destructible   bool
+    Creator        string
+}
+```
+
+This message is expected to fail if:
+
+- `LpTokenDenom` does not comply with the rules specified on the chain.
+- the name of farm pool has exist.
+- `StartHeight` is less than the current block height.
+- `TotalReward` is less than `RewardPerBlock`.
+- the balance of creator is not enough to pay `CreatePoolFee+TotalReward`.
+- The lenght of `TotalReward` is greater than `MaxRewardCategoryN`.
+
+In addition, the `Endheight = TotalReward/RewardPerBlock`.Because there may be multiple tokens for event rewards, the end heights may be inconsistent. In order to reduce the complexity of the system, take the smallest value among all heights as the final end height. After the event ends, the remaining rewards will be refunded To creator's account.
+
+At the beginning of the activity, because there was no user participating, so `RewardPerShare=0`, which means that the user has no income from the beginning of the activity to the user's first stake, and every time the user's `stake`、 `unstake` 、`harvest` will trigger
+in the calculation of `RewardPerShare` (calculate the income that each lptoken can obtain before), the user's previous income is equal to `lastTotalLocked*RewardPerShare-lastDebt`, after the user gets back the income, record the user's current total debt (total income that has been withdrawn, lastDebt) is equal to `currentTotalLocked*RewardPerShare`.
+
+## MsgDestroyPool
+
+The farm pool can be destroyed by creator via a `MsgDestroyPool` message.
+
+```go
+type MsgDestroyPool struct {
+    PoolName string
+    Creator  string
+}
+```
+
+This message is expected to fail if:
+
+- the farm pool is not exist.
+- the `Creator` is not the creator of the farm pool.
+- the farm pool activity has ended.
+
+When creator actively destroys the pool, it will also trigger the recalculation of `RewardPerShare`, calculate the current value of each `lpToken` (relative to the reward), and calculate the current remaining bonus in the pool, and return the bonus to creator to remove the pool from the active queue.
+
+When the pool becomes inactive (or expired、 ended), the user can only unstake his staking `lptoken` from the pool and get rewards, or it can be retrieved multiple times, but the reward obtained are the same as all at once. note that `RewardPerShare` will no longer be updated at this time
```

Comment:
```suggestion
When the pool becomes inactive (expired or ended), the user can only unstake his staking `lptoken` from the pool and get rewards, or it can be retrieved multiple times, but the reward obtained are the same as all at once. note that `RewardPerShare` will no longer be updated at this time
```